compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETargetDirect.cpp

```diff
@@ -0,0 +1,417 @@
+      if (ecode) {
+        return moduleOp.emitError()
+               << "failed to create working directory " << workDir
+               << ". Error message : " << ecode.message();
+      }
+      return workDir;
+    }
+
+    // No path for intermediates: make a temporary directory for this
+    // executable that is certain to be distinct from the dir of any other
+    // executable.
+    SmallString<128> workDirFromScratch;
+    auto err = llvm::sys::fs::createUniqueDirectory(
+        /* prefix = */ variantOp.getName(), workDirFromScratch);
+
+    if (err)
+      return moduleOp.emitOpError()
+             << "failed to create working directory for xclbin generation: "
+             << err.message();
+
+    return workDirFromScratch;
+  }();
+
+  if (failed(maybeWorkDir)) return failure();
+  auto workDir = maybeWorkDir.value();
+
+  // collect names of kernels as they need to be in kernels.json
+  // generated by `aie2xclbin`
+  SmallVector<std::string> entryPointNames;
+  // collect the aie.device ops that need to be passed to the `aie2xclbin` tool
+  SmallVector<xilinx::AIE::DeviceOp> deviceOps;
+  // Map to keep track of which ordinal number belongs to which entry point,
+  // typically the order is sequential but that is not gauranteed
+  std::map<std::string, uint64_t> entryPointOrdinals;
+  for (auto exportOp : variantOp.getExportOps()) {
+    uint64_t ordinal = 0;
+    if (std::optional<APInt> optionalOrdinal = exportOp.getOrdinal()) {
+      ordinal = optionalOrdinal->getZExtValue();
+    } else {
+      // For executables with only one entry point, linking doesn't kick in at
+      // all. So the ordinal can be missing for this case.
+      if (!llvm::hasSingleElement(variantOp.getExportOps())) {
+        return exportOp.emitError() << "should have ordinal attribute";
+      }
+    }
+
+    StringRef exportOpName = exportOp.getSymName();
+    deviceOps.push_back(getDeviceOpFromEntryPoint(moduleOp, exportOpName));
+
+    // The xclbin kernel name, appended with instance name suffix (`:MLIRAIEV1`,
+    // 10 chars) is required by the xclbinutil to have a length smaller or equal
+    // to 64 right now. To have some additional wiggle room for suffix changes,
+    // we use the 42 first characters for the kernel name. We suffix ordinal
+    // number to the name to gaurantee uniqueness within the executable.
+    std::string entryPointName = llvm::join_items(
+        "_", exportOpName.substr(0, 42), std::to_string(ordinal));
+    entryPointNames.emplace_back(entryPointName);
+    entryPointOrdinals[entryPointName] = ordinal;
+    // error out if we think the name will most likely be too long
+    // for the artifact generation to succeed. We set this cut-off at 50
+    // characters.
+    if (entryPointName.size() > 50)
+      return exportOp.emitError()
+             << "entry point name: " << entryPointName << "is too long!";
+  }
+  uint64_t ordinalCount = entryPointOrdinals.size();
+
+  std::vector<uint32_t> npuInstrs;
```

Comment:
This is a bug I introduced sorry,

https://github.com/nod-ai/iree-amd-aie/pull/453